src/dmd/link.d

```diff
@@ -32,6 +32,27 @@ version (Windows) extern (C) int putenv(const char*);
 version (Windows) extern (C) int spawnlp(int, const char*, const char*, const char*, const char*);
 version (Windows) extern (C) int spawnl(int, const char*, const char*, const char*, const char*);
 version (Windows) extern (C) int spawnv(int, const char*, const char**);
+
+string getenv2(const(char)* name)
+{
+    auto result = getenv(name);
+    return cast(string)result[0 .. strlen(result)];
+}
+int putenv2(const char[] name, const(char)* value)
+{
+    return putenv2(name, value[0 .. strlen(value)]);
+}
+int putenv2(const char[] name, const char[] value)
+{
+    auto fullLength = name.length + 1 + value.length;
+    auto buffer = cast(char*)alloca(fullLength + 1);
+    buffer[0               .. name.length] = name[];
+    buffer[name.length]                    = '=';
+    buffer[name.length + 1 .. fullLength ] = value[];
+    buffer[fullLength] = '\0';
+    return putenv(buffer);
+}
```

Comment:
Imho `{get,put}env2` is a genuinely bad name for functions for D-ified versions.
Maybe `dgetenv` or simply use overloading?